x-pack/plugins/security_solution/public/detection_engine/rule_gaps/components/manual_rule_run/index.test.tsx

```diff
@@ -6,73 +6,78 @@
  */
 
 import React from 'react';
-import { render, within } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import { ManualRuleRunModal } from '.';
 
+const DATE_PICKER_PREVIOUS_BTN_CLASS = '.react-datepicker__navigation--previous';
+const DATE_PICKER_NEXT_BTN_CLASS = '.react-datepicker__navigation--next';
+
 describe('ManualRuleRunModal', () => {
   const onCancelMock = jest.fn();
   const onConfirmMock = jest.fn();
 
+  let startDatePicker: HTMLElement;
+  let endDatePicker: HTMLElement;
+  let confirmModalConfirmButton: HTMLElement;
+  let cancelModalConfirmButton: HTMLElement;
+  let timeRangeForm: HTMLElement;
+
   afterEach(() => {
     onCancelMock.mockReset();
     onConfirmMock.mockReset();
   });
 
-  it('should render modal', () => {
-    const wrapper = render(
-      <ManualRuleRunModal onCancel={onCancelMock} onConfirm={onConfirmMock} />
-    );
+  beforeEach(() => {
+    render(<ManualRuleRunModal onCancel={onCancelMock} onConfirm={onConfirmMock} />);
+
+    startDatePicker = screen.getByTestId('start-date-picker');
+    endDatePicker = screen.getByTestId('end-date-picker');
+    confirmModalConfirmButton = screen.getByTestId('confirmModalConfirmButton');
+    cancelModalConfirmButton = screen.getByTestId('confirmModalCancelButton');
+    timeRangeForm = screen.getByTestId('manual-rule-run-time-range-form');
+  });
 
-    expect(wrapper.getByTestId('manual-rule-run-modal-form')).toBeInTheDocument();
-    expect(wrapper.getByTestId('confirmModalCancelButton')).toBeEnabled();
-    expect(wrapper.getByTestId('confirmModalConfirmButton')).toBeEnabled();
+  it('should render modal', () => {
+    expect(timeRangeForm).toBeInTheDocument();
+    expect(cancelModalConfirmButton).toBeEnabled();
+    expect(confirmModalConfirmButton).toBeEnabled();
   });
 
   it('should render confirmation button disabled if invalid time range has been selected', () => {
-    const wrapper = render(
-      <ManualRuleRunModal onCancel={onCancelMock} onConfirm={onConfirmMock} />
-    );
+    render(<ManualRuleRunModal onCancel={onCancelMock} onConfirm={onConfirmMock} />);
```

Comment:
Are these subsequent `render` calls necessary?

If the elements are assigned in the `beforeEach`, subsequent `render`s aren't going to update those references, right?

Reply:
No these renders should be removed! I'm not sure why those references still work though